projects/vdk-core/tests/functional/run/test_run_standalone_data_job.py

```diff
@@ -1,15 +1,62 @@
 # Copyright 2021 VMware, Inc.
 # SPDX-License-Identifier: Apache-2.0
+import collections
 from pathlib import Path
 
+import pytest
 from functional.run import util
 from vdk.internal.builtin_plugins.run.standalone_data_job import (
     StandaloneDataJobFactory,
 )
 
+ExecutedStandaloneDataJobDetails = collections.namedtuple(
+    "ExecutedStandalongDataJob", "hook_tracker job_input_name"
+)
+
+
+@pytest.fixture(scope="module")
+def executed_standalone_data_job():
+    hook_tracker = util.HookCallTracker()
+    job_input_name = "unknown"
+
+    with StandaloneDataJobFactory.create(
+        data_job_directory=Path(__file__), extra_plugins=[hook_tracker]
+    ) as job_input:
+        job_input_name = job_input.get_name()
+
+    return ExecutedStandaloneDataJobDetails(
+        hook_tracker=hook_tracker,
+        job_input_name=job_input_name,
+    )
+
+
+def test_standalone_data_job_named_after_data_job_directory(
+    executed_standalone_data_job,
+):
+    assert executed_standalone_data_job.job_input_name == Path(__file__).name
+
+
+def test_standalone_data_job_executes_hooks_in_expected_order(
+    executed_standalone_data_job,
+):
+    assert (
+        ", ".join(executed_standalone_data_job.hook_tracker.calls)
+        == "vdk_start, vdk_configure, vdk_initialize, initialize_job, run_job, run_step, finalize_job"
+    )
+
+
+@pytest.mark.skip(
+    reason="Not sure whether vdk_exit should be called, given we're not executing from the command line"
+)
+def test_standalone_data_job_executes_vdk_exit_hooks(executed_standalone_data_job):
+    assert "vdk_exit" in executed_standalone_data_job.hook_tracker.calls
+
```

Comment:
@tozka, Do you think the `StandaloneDataJob` should call the `vdk_exit` hook? 

I'm unsure from the [docs](https://github.com/vmware/versatile-data-kit/blob/main/projects/vdk-core/src/vdk/api/plugin/core_hook_spec.py#L113) where this is CLI specific behaviour

Reply:
Yes, I bet there are (and will be) plugins providing some functionality for data jobs that rely on vdk_exit.